src/routes/auth.js

```diff
@@ -600,23 +600,48 @@ async function changePassword(request, h) {
 }
 
 async function resendActivation(request, h) {
-    const { email } = get(request, ['auth', 'artifacts'], {});
     const isAdmin = request.server.methods.isAdmin(request);
     const { domain, https } = request.server.methods.config('frontend');
 
-    if (!isAdmin && request.payload.email !== email) {
-        return Boom.forbidden();
+    const email =
+        isAdmin && request.payload && request.payload.email
+            ? request.payload.email
+            : get(request, ['auth', 'artifacts', 'email'], {});
+
+    if (!email) {
```

Comment:
The `if (!email) {}` block will never get called. The third argument to `get` is a default value which means if `request.auth.artifacts.email` is `undefined` you set it to an empty object `{}` which is always truthy `Boolean({}) === true`.

This suggestion should solve the issue:
```suggestion
    const email =
        isAdmin && get(request, ['payload', 'email'])
            ? request.payload.email
            : get(request, ['auth', 'artifacts', 'email']);

    if (!email) {
```

Yay for multi line suggestions! 🎉

Reply:
Actually it will get called if `request.auth.artifacts.email = null`. Which should never be the case.